string regul_name
string name
float64 value